app/controllers/admin_only/member_app_waiting_reasons_controller.rb

```diff
@@ -0,0 +1,92 @@
+module AdminOnly
+
+  class MemberAppWaitingReasonsController < ApplicationController
+
+    before_action :set_member_app_waiting_reason, only: [:show, :edit, :update, :destroy]
+    before_action :authorize_member_app_waiting_reason, only: [:update, :show, :edit]
```

Comment:
Yep -- you're right that `new` and `create` need the authorization.   But the way to do the authorization is different than in the methods listed here (because we need to use the _class_ to check, since we do not yet have an instance to work with).

👍